generics/services/sunbird.js

```diff
@@ -0,0 +1,177 @@
+        } else {
+            bodyData["request"]["limit"] = dataLimit;
+        }
+
+        if ( pageNo !== "" ) {
+            let offsetValue = pageSize * ( pageNo - 1 ); 
+            bodyData["request"]["offset"] = offsetValue;
+        }
+
+        if ( searchKey !== "" ) {
+            bodyData["request"]["query"] = searchKey
+        }
+
+        const url = 
+        sunbirdBaseUrl + messageConstants.endpoints.GET_LOCATION_DATA;
+        const options = {
+            headers : {
+                "Authorization" : process.env.SUNBIRD_SERVICE_AUTHERIZATION,
+                "content-type": "application/json"
+            },
+            json : bodyData
+        };
+
+        request.post(url,options,locationSearchCallback);
+
+        let result = {
+            success : true
+        };
+
+        function locationSearchCallback(err, data) {
+
+            if (err) {
+                result.success = false;
+            } else {
+                  
+                let response = data.body;
+                if ( response.responseCode === messageConstants.common.OK ) {
+                    result["data"] = response.result;
+                } else {
+                      result.success = false;
+                }
+            }
+            return resolve(result);
+        }
+
+        setTimeout( function () {
+            return reject (result = {
+                success : false
+             });
+         }, timeout);
+
+      } catch (error) {
+          return reject(error);
+      }
+  })
+}
+
+/**
+  * 
+  * @function
+  * @name orgSchoolSearch
+  * @param {String} bearerToken - autherization token.
+  * @param {object} bodyData -  location id
+  * @returns {Promise} returns a promise.
+*/
+const orgSchoolSearch = function ( filterData, pageSize = "", pageNo = "", searchKey = "", fields) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            let bodyData = {};
+            bodyData["request"] = {};
+            bodyData["request"]["filters"] = filterData;
+
+            if ( pageSize !== "" ) {
+                bodyData["request"]["limit"] = pageSize;
+            } 
+    
+            if ( pageNo !== "" ) {
+                let offsetValue = pageSize * ( pageNo - 1 ); 
+                bodyData["request"]["offset"] = offsetValue;
+            }
+    
+            if ( searchKey !== "" ) {
+                bodyData["request"]["fuzzy"] = {
+                    "orgName" : searchKey
+                }
+            }
+
+            //for getting specified key data only.
+            if ( fields.length > 0 ) {
+                bodyData["request"]["fields"] = fields;
+            }
+            
+            
+            
+            const url = 
+            sunbirdBaseUrl + messageConstants.endpoints.GET_SCHOOL_DATA;
+            const options = {
+                headers : {
+                    "Authorization" : process.env.SUNBIRD_SERVICE_AUTHERIZATION,
```

Comment:
@VISHNUDAS-tunerlabs do we require Authorization in the internal call ? check user profile API calls